### PR TITLE
Week11/aws beta hosting

### PR DESCRIPTION
**Pull request by adari06**
# Pull Request

## Summary

Briefly describe what changed.

- What was added/modified?
- What problem does this address?

## Why

Explain the motivation behind this change.

- Is this part of a weekly deliverable?
- Does this address a bug or architectural improvement?

## How to Verify

Provide clear steps to verify this change locally.

1. Install dependencies: `npm install`
2. Run the app / relevant command
3. Run lint: `npm run eslint`
4. Run tests: `npm run test`
5. Describe expected behavior

## Checklist (Definition of Done)

- Branch follows team naming pattern (`week<NUM>/<short-description>`)
- PR opened early and kept reviewable in size
- At least one teammate review requested
- CI checks are passing
- Lint passes locally (`npm run eslint`)
- Tests pass locally (`npm run test`)
- New behavior includes starter tests or test plan notes
- Documentation updated if needed (`README`, `docs/`, ADRs)

## Notes for Reviewers

Anything specific you'd like feedback on?
- Edge cases?
- Architecture concerns?
- Naming?
- Test coverage?

### CODE REVIEW

File: .platform/hooks/predeploy/01_mount_ebs.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+# Attaches and mounts the persistent EBS volume that holds the SQLite database.
+# Runs as root during confighooks/predeploy on every deployment.
```
Comment (hmhawn) [inline .platform/hooks/predeploy/01_mount_ebs.sh]: replace with: # Runs as root from .platform/hooks/predeploy on every deployment.

File: .platform/hooks/predeploy/02_migrate.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Runs Knex database migrations against the production SQLite database.
+# Runs as the webapp service user during hooks/predeploy on every deployment,
```
Comment (hmhawn) [inline .platform/hooks/predeploy/02_migrate.sh]: replace with: # Runs under the Elastic Beanstalk predeploy hook user (root by default).

File: server.js [hunk 3]
```diff
@@ -25,9 +27,63 @@ const swaggerDocument = YAML.load('./docs/api/openapi.yaml');
...
+    return {
+      path: dbPath,
+      writable: false,
+      target: targetPath,
+      error: error.message,
+    };
+  }
+}
+
+function ensureProductionStorageReady() {
```
Comment (hmhawn) [inline server.js]: Current behavior:

Only checks if path is writable
DOES NOT confirm /data is actually mounted EBS

REQUIRED FIX: Add mount check before writable check

import fs from 'fs';

function isMounted(targetPath) {
  try {
    const mounts = fs.readFileSync('/proc/mounts', 'utf8');
    return mounts.includes(targetPath);
  } catch {
    return false;
  }
}

**Then update function:**

function ensureProductionStorageReady() {
  if (!isProduction) return;

  if (!process.env.SQLITE_PATH) {
    throw new Error('SQLITE_PATH must be set in production.');
  }

  if (process.env.SQLITE_PATH.startsWith('/data')) {
    if (!isMounted('/data')) {
      throw new Error('/data is not mounted. Refusing to start.');
    }
  }

  const dbStatus = getDbPathStatus();
  if (!dbStatus.writable) {
    throw new Error(`Configured SQLite path is not writable: ${dbStatus.target}`);
  }
}

**Comment by adari06** [comment]
**Summary of fixes applied to address all review comments:**

 **Production storage safety**: Added isMounted() function to verify /data is actually mounted (not just writable) before starting in production. This prevents silent writes to ephemeral disk if EBS mount fails.

 **Health endpoint security**: Reduced information disclosure in production - /health now returns only {writable: boolean} instead of full filesystem paths and error details.

 **EB environment config**: 
- Added SQLITE_PATH: /data/trips.db
- Added EBS_VOLUME_ID: REPLACE_IN_EB_ENV placeholder for per-environment configuration
- Removed insecure placeholder secrets (GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, SESSION_SECRET) with note that these must be set via EB environment properties

 **Script header corrections**: 
- Fixed hook location references to match actual paths (confighooks/predeploy  .platform/hooks/predeploy)
- Corrected execution user documentation for predeploy scripts

 **Duplicate mount script**: Converted .platform/confighooks/predeploy/01_mount_ebs.sh to a no-op with deprecation notice, preventing double-execution and future divergence.

 **Documentation accuracy**: Updated aws-hosting-plan.md to reflect that the file is versioned for team visibility.

Ready for re-review!

File: data/aws-hosting-implementation.md
Comment (hmhawn) [inline data/aws-hosting-implementation.md]: This is in much better shape now and the main review issues look addressed.

One thing I would still update before approval: `data/aws-hosting-implementation.md` is now partially out of sync with the implementation. It still describes `/health` as returning DB path/details, but the code was intentionally changed to avoid exposing those in production. It also still lists a few items as open that appear to have been implemented in this PR (startup mount guard, session/cookie hardening, attach/mount automation).

I’d recommend updating that doc so it reflects the current implementation state before merge.
